test/README.md

```diff
@@ -0,0 +1,179 @@
+
+# Test
+
+This directory contains tests and testing docs for the knative-sandbox `eventing-kafka` project.
+
+- [Unit Tests](#unit-tests) reside in the codebase alongside the code under test.
+- [Integration Tests](#integration-tests) consist of the `conformance` and `e2e` tests also in this directory.
+- [Performance Tests](#performance-tests) have yet to be completed but will reside here as well.
+
+While some information has been repeated here, it is useful to be familiar with the other
+Knative testing documentation such as...
+
+- [Serving](https://github.com/knative/serving/blob/master/test/README.md)
+- [Eventing](https://github.com/knative/eventing/tree/master/test/README.md)
+- [Eventing-Contrib](https://github.com/knative/eventing-contrib/blob/master/test/README.md)
+
+## Unit Tests
+
+The Unit Tests are run by Prow for all Pull Requests.  They are executed by Prow via the
+[pre-submit-tests.sh](./presubmit-tests.sh) script with the `--unit-tests` argument.
+Or you can always run them manually via the `go test` cmd line...
+
+```
+go test -v pkg/channel/distributed/...
+```
+
+## Integration Tests
+
+The `End-To-End` and `Conformance` tests are Knative "integration-tests" meant to be run by Prow for
+all Pull Requests.  They are executed by Prow via the [pre-submit-tests.sh](./presubmit-tests.sh)
+script with the `--integration-tests` argument.  The existence of the [e2e-tests.sh](./e2e-tests.sh)
+script will cause them to be run in that use case.
+
+For local development / maintenance of the tests however, it is convenient to be able to run them
+in various manners which are described here.
+
+### Pre-Submit Script
+
+The [`presubmit-tests.sh`](./presubmit-tests.sh) script is the entry point for the tests run
+from Prow when a PR is created.
+
+By default, the script will run `build tests`, `unit tests` and `integration tests`.  If you
+only want to run one type of tests, you can run this script with corresponding flags such as...
+
+```
+./test/presubmit-tests.sh
+./test/presubmit-tests.sh --build-tests
+./test/presubmit-tests.sh --unit-tests
+./test/presubmit-tests.sh --integration-tests
+```
+
+The script will call the [`e2e-tests.sh`](./e2e-tests.sh) script if it exists.
+
+### E2E-Tests Script
+
+The [`e2e-tests.sh`](./e2e-tests.sh) script is the entry point for running all the e2e tests.
+
+By default, it will create a new GKE cluster in project `$PROJECT_ID`, start Knative Eventing,
+upload test images to `$KO_DOCKER_REPO`, and run the `conformance` and `e2e` tests. After the
+tests finishes, it will delete the cluster.  To skip cluster teardown for diagnostic purposes,
+"--skip-teardowns" can be added to the script arguments or to the initialize command in
```

Comment:
<!--mdfmt-bot-->
Format markdown:
```suggestion
By default, it will create a new GKE cluster in project `$PROJECT_ID`, start
Knative Eventing, upload test images to `$KO_DOCKER_REPO`, and run the
`conformance` and `e2e` tests. After the tests finishes, it will delete the
cluster. To skip cluster teardown for diagnostic purposes, "--skip-teardowns"
can be added to the script arguments or to the initialize command in
```